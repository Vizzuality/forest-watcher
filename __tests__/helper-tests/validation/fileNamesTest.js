import 'react-native';

import fileNameIsValid from '../../../app/helpers/validation/fileNames';

const exampleFiles = [
  {
    id: '1',
    uri: 'testuri',
    path: 'testpath',
    type: 'testtype',
    fileName: 'testfilename',
    size: 123,
    name: 'Test1'
  },
  {
    id: '2',
    uri: 'testuri',
    path: 'testpath',
    type: 'testtype',
    fileName: 'testfilename',
    size: 123,
    name: 'Test2'
  },
  {
    id: '3',
    uri: 'testuri',
    path: 'testpath',
    type: 'testtype',
    fileName: 'testfilename',
    size: 123,
    name: 'Test3'
  }
];

describe('fileNames', () => {
  describe('func fileNameIsValid', () => {
    describe('when no files already exist', () => {
      it('returns invalid when provided with a null name', () => {
        expect(fileNameIsValid('4', null, [], 5)).toMatchSnapshot();
      });

      it('returns valid when provided with a file that has a short name', () => {
        expect(fileNameIsValid('4', 'A', [], 5)).toMatchSnapshot();
      });

      it('returns valid when provided with a file name just below the length limit', () => {
        expect(fileNameIsValid('4', 'Test', [], 5)).toMatchSnapshot();
      });

      it('returns valid when provided with a file name on the length limit', () => {
        expect(fileNameIsValid('4', 'Test4', [], 5)).toMatchSnapshot();
      });

      it('returns invalid when provided with a file name just above the length limit', () => {
        expect(fileNameIsValid('4', 'Test-4', [], 5)).toMatchSnapshot();
      });

      it('returns invalid when provided with a file name above the length limit', () => {
        expect(fileNameIsValid('4', 'Test--4', [], 5)).toMatchSnapshot();
      });
    });

    describe('when files already exist', () => {
      it('returns valid when the file & name do not already exist', () => {
        expect(fileNameIsValid('4', 'Test4', exampleFiles, 5)).toMatchSnapshot();
      });

      // Tests an edge-case where if we attempt to give an existing file its previous name,
      // we shouldn't mark it as invalid.
      it('returns valid when the file exists & the name matches', () => {
        expect(fileNameIsValid('3', 'Test3', exampleFiles, 5)).toMatchSnapshot();
      });

      it('returns invalid when the file does not exist & the name matches', () => {
        expect(fileNameIsValid('4', 'Test3', exampleFiles, 5)).toMatchSnapshot();
      });
    });
  });
});
