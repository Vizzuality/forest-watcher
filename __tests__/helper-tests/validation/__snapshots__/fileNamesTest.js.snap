// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fileNames func fileNameIsValid when files already exist returns invalid when the file does not exist & the name matches 1`] = `
Object {
  "alreadyTaken": true,
  "valid": false,
}
`;

exports[`fileNames func fileNameIsValid when files already exist returns valid when the file & name do not already exist 1`] = `
Object {
  "alreadyTaken": false,
  "valid": true,
}
`;

exports[`fileNames func fileNameIsValid when files already exist returns valid when the file exists & the name matches 1`] = `
Object {
  "alreadyTaken": false,
  "valid": true,
}
`;

exports[`fileNames func fileNameIsValid when no files already exist returns invalid when provided with a file name above the length limit 1`] = `
Object {
  "alreadyTaken": false,
  "valid": false,
}
`;

exports[`fileNames func fileNameIsValid when no files already exist returns invalid when provided with a file name just above the length limit 1`] = `
Object {
  "alreadyTaken": false,
  "valid": false,
}
`;

exports[`fileNames func fileNameIsValid when no files already exist returns invalid when provided with a null name 1`] = `
Object {
  "alreadyTaken": false,
  "valid": false,
}
`;

exports[`fileNames func fileNameIsValid when no files already exist returns valid when provided with a file name just below the length limit 1`] = `
Object {
  "alreadyTaken": false,
  "valid": true,
}
`;

exports[`fileNames func fileNameIsValid when no files already exist returns valid when provided with a file name on the length limit 1`] = `
Object {
  "alreadyTaken": false,
  "valid": true,
}
`;

exports[`fileNames func fileNameIsValid when no files already exist returns valid when provided with a file that has a short name 1`] = `
Object {
  "alreadyTaken": false,
  "valid": true,
}
`;
